src/app/(sidebar)/account/fund/page.tsx

```diff
@@ -1,5 +1,185 @@
 "use client";
 
+import { useState } from "react";
+import Link from "next/link";
+
+import {
+  Card,
+  Icon,
+  IconButton,
+  Input,
+  Text,
+  Button,
+} from "@stellar/design-system";
+
+import { shortenStellarAddress } from "@/helpers/shortenStellarAddress";
+import { validatePublicKey } from "@/helpers/validatePublicKey";
+import { useStore } from "@/store/useStore";
+
+import { AlertBox } from "@/components/AlertBox";
+
+import "../styles.scss";
+
+const callFriendBot = async ({
+  network,
+  publicKey,
+}: {
+  network: string;
+  publicKey: string;
+}) => {
+  const friendbotURL =
+    network === "futurenet"
+      ? "https://friendbot-futurenet.stellar.org"
+      : "https://friendbot.stellar.org";
+  const response = await fetch(friendbotURL + "/?addr=" + publicKey);
+
+  return response;
+};
+
 export default function FundAccount() {
-  return <div>Fund Account</div>;
+  const { account, network } = useStore();
+
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const [generatedPublicKey, setGeneratedPublicKey] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const fundAccount = async () => {
+    try {
+      setIsLoading(true);
+      setError("");
+
+      const data = await callFriendBot({
+        network: network.id,
+        publicKey: account.publicKey,
+      });
+
+      if (!data.ok) {
+        const errorBody = await data.json();
+
+        throw new Error(errorBody.status || "An error occurred");
+      }
+
+      setIsSuccess(true);
+    } catch (error: any) {
+      let errorMessage = "";
+
+      if (error.status === 0) {
+        errorMessage = `Unable to reach Friendbot server at ${network.id}`;
+      } else {
+        errorMessage = `Failed to fund ${account.publicKey} on the ${network.id}`;
+      }
+      setError(errorMessage);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <div className="Account">
+      <Card>
+        <div className="Account__card">
+          <div className="CardText">
+            <Text size="lg" as="h1" weight="medium">
+              Friendbot: fund a {network.id} network account
+            </Text>
+
+            <Text size="sm" as="p">
+              The friendbot is a horizon API endpoint that will fund an account
+              with 10,000 lumens on the {network.id} network.
+            </Text>
+          </div>
+
+          <Input
+            id="generate-keypair-publickey"
+            fieldSize="md"
+            label="Public Key"
+            value={generatedPublicKey}
+            onChange={(e) => {
+              setGeneratedPublicKey(e.target.value);
```

Comment:
We should validate the public key on change for better UX. And enable submit only if it is valid. That way, we will avoid unnecessary failures. What do you think?

Reply:
@quietbits, I set validating key `onClick` [here](https://github.com/stellar/laboratory/pull/788/files#diff-089b35fceb1617e09abad820d0b3e6a3036a1f51264f29a582013d102d9b3f23R114-R122) before submitting to the network.